packages/clients/src/aptos/wallet.ts

```diff
@@ -0,0 +1,431 @@
+/**
+ * Copyright 2023-2024 Shinami Corp.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SimpleTransaction,
+  MultiAgentTransaction,
+  AnyRawTransaction,
+  AccountAuthenticator,
+  PendingTransactionResponse,
+} from "@aptos-labs/ts-sdk";
+import { Infer, array, integer, object, string, unknown } from "superstruct";
+import { ShinamiRpcClient, errorDetails, trimTrailingParams } from "../rpc.js";
+import { JSONRPCError } from "@open-rpc/client-js";
+
+const KEY_RPC_URL = "https://api.shinami.com/aptos/key/v1";
+const WALLET_RPC_URL = "https://api.shinami.com/aptos/wallet/v1";
+
+/**
+ * Shinami Key RPC client for Aptos.
+ */
+export class KeyClient extends ShinamiRpcClient {
+  /**
+   * @param accessKey Aptos Wallet access key.
+   * @param url Optional URL override.
+   */
+  constructor(accessKey: string, url: string = KEY_RPC_URL) {
+    super(accessKey, url);
+  }
+
+  /**
+   * Creates a session with the provided wallet secret.
+   * @param secret Wallet secret. For later operations to succeed, the secret must match what the
+   *    wallet was created with.
+   * @returns A session token used to perform sensitive wallet operations with, valid for 10 mins.
+   */
+  createSession(secret: string): Promise<string> {
+    return this.request("key_createSession", [secret], string());
+  }
+}
+
+/**
+ * Data wrapper for a Shinami Aptos wallet.
+ */
+const WalletInfo = object({
+  accountAddress: string(),
```

Comment:
There has been a design philosophy shift in Aptos gas client that we are integrating more closely with Aptos sdk using native types. I suggest making this `AccountAddress` because it's returned to callers (you should also `export` the type), and you can create a sister type `GetWalletResult` for internal JSON RPC typing.

Reply:
Made this more in line with returning native Aptos sdk types. One small deviation from your suggestion: I used `WalletResult` as the internal json rpc type since it's used in the wallet operation methods (create, initialize), not just `getWallet`

Reply:
For consistency, name this `GetWalletResult`.

Reply:
resolved offline. using `WalletResult`